harness-delegate-ng/values.yaml

```diff
@@ -97,3 +97,29 @@ securityContext:
 
 nextGen: true
 
+# Configure a Kubernetes build farm to use self-signed certificates
+# https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/configure-a-kubernetes-build-farm-to-use-self-signed-certificates/
+shared_certificates:
+  enabled: false
+  certs_path: /shared/customer-artifacts/certificates/ca.bundle
+  # Example Certificate Chain (Multi-line files )
+  # ca_bundle should be the text of the CA Bundle to include in a secret
+  ca_bundle: # |
+  #   -----BEGIN CERTIFICATE-----
+  #   XXXXXXXXXXXXXXXXXXXXXXXXXXX
+  #   -----END CERTIFICATE-------
+  #   -----BEGIN CERTIFICATE-----
+  #   XXXXXXXXXXXXXXXXXXXXXXXXXXX
+  #   -----END CERTIFICATE-------
+
+  # CI Mount targets are the locations that the secrets should be mounted in the CI Images
+  ci_mount_targets:
+    # - /etc/ssl/certs/ca-bundle.crt
```

Comment:
The suggested mount points clash with the system locations where these will get mounted, which will cut off the standard set of public certs normally located in the trust store for that image.  All certs that need to be trusted for those containers would need specified in the values file.

I think it would be worth calling this out in the comments like the page you link to does (https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/configure-a-kubernetes-build-farm-to-use-self-signed-certificates/).  Ideally it would be even better if this list could be layered on top of the standard set of trusted certs, but that change would likely be outside of the scope of just this helm chart.

Reply:
From what I understand, the certificate is mounted directly not as a folder but as a file.  This would allow it to coexist with other trust stores in the image.  Now, I guess the document assumes that the customer provided certchain includes details about to level certificate trust chains as well.   I can only assume that the documentation is correct in this case.

Reply:
That file is already present in some images, one example being `/etc/ssl/certs/ca-bundle.crt` in `harness/drone-git`.  If a user overwrites that to clone from their GHA signed with an private CA cert, then they will no longer be able to clone from github.com without adding DigiCert to the list as well.

Reply:
Added the same warning about replacing the default trust chain details from the linked document into the Values.yaml as a comment

>CAUTION
Make sure that the destination path is not same as the default CA certificate path of the corresponding container image.
>
>If you want to override the default certificate file, make sure the Kubernetes secret or config map (from step one) includes all certificates required by the pipelines that will use this build infrastructure.